ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -107,70 +121,137 @@ public void execute(Input input, ActionListener<Output> listener) {
         List<ModelTensor> modelTensors = new ArrayList<>();
         outputs.add(ModelTensors.builder().mlModelTensors(modelTensors).build());
 
+        FetchSourceContext fetchSourceContext = new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
+        GetDataObjectRequest getDataObjectRequest = GetDataObjectRequest
+            .builder()
+            .index(ML_AGENT_INDEX)
+            .id(agentId)
+            .tenantId(tenantId)
+            .fetchSourceContext(fetchSourceContext)
+            .build();
+
         if (clusterService.state().metadata().hasIndex(ML_AGENT_INDEX)) {
             try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-                GetRequest getRequest = new GetRequest(ML_AGENT_INDEX).id(agentId);
-                client.get(getRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
-                    if (r.isExists()) {
-                        try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
-                            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-                            MLAgent mlAgent = MLAgent.parse(parser);
-                            MLMemorySpec memorySpec = mlAgent.getMemory();
-                            String memoryId = inputDataSet.getParameters().get(MEMORY_ID);
-                            String parentInteractionId = inputDataSet.getParameters().get(PARENT_INTERACTION_ID);
-                            String regenerateInteractionId = inputDataSet.getParameters().get(REGENERATE_INTERACTION_ID);
-                            String appType = mlAgent.getAppType();
-                            String question = inputDataSet.getParameters().get(QUESTION);
-
-                            if (memoryId == null && regenerateInteractionId != null) {
-                                throw new IllegalArgumentException("A memory ID must be provided to regenerate.");
+                sdkClient
+                    .getDataObjectAsync(getDataObjectRequest, client.threadPool().executor("opensearch_ml_general"))
```

Comment:
the executor should probably be a constant somewhere?

Reply:
I'm thinking to move `GENERAL_THREAD_POOL` (along with other POOLs static variable) to Common module so that algorithm & plugin both can have access. But that demands a separate PR. So it'll here for a while